src/YobotBrokerV2.sol

```diff
@@ -0,0 +1,140 @@
+        Drop storage drop = drops[_dropId];
+        require(isLocked(drop), 'drop not locked');
+        uint128 newLockLength = uint128(block.timestamp) - drop.lockStartTime;
+        drop.lockLength = newLockLength;
+
+        emit DropInfo(_dropId, drop.token, drop.lockStartTime, newLockLength);
+    }
+
+    function fillOrder(uint256 _dropId, address _user, uint256 _tokenId) external onlyOperator {
+        Drop storage drop = drops[_dropId];
+        Order storage order = dropIdToUserOrder[_dropId][_user];
+
+        // EFFECTS
+        // memoization
+        uint128 amountDeposited = order.amountDeposited;
+        uint128 quantity = order.quantity;
+        uint128 priceInWeiEach = amountDeposited / quantity;
+        // state updates
+        order.quantity -= 1; // reverts on underflow
+		order.amountDeposited -= priceInWeiEach;
+		yoBotBalance += priceInWeiEach; // pay the bot
+        // events
+		emit OrderFilled(_dropId, _user, _tokenId);
+		emit OrderInfo(_dropId, _user, quantity - 1, amountDeposited - priceInWeiEach);
+
+		// INTERACTIONS
+		// transfer NFT to user
+		IERC721(drop.token).safeTransferFrom(msg.sender, _user, _tokenId); // ERC721-compliant contracts revert on failure here
+    }
+
+    function collectPayment() external onlyOperator {
+        uint256 amountToCollect = yoBotBalance;
+        yoBotBalance = 0;
+        sendValue(payable(msg.sender), amountToCollect);
+    }
+
+    // USER FUNCTIONS
+
+    function createNewOrder(uint256 _dropId, uint128 _quantity) external payable {
+        Drop storage drop = drops[_dropId];
+        require(dropIdToUserOrder[_dropId][msg.sender].amountDeposited == 0, 'you already have an order for this drop');
+        require(isBeforeLockStartTime(drop), 'orders are locked for this drop');
+        require(_quantity > 0, '0 quantity orders not accepted');
+        require(msg.value > 0, 'no free NFTs');
+        dropIdToUserOrder[_dropId][msg.sender].quantity = _quantity;
+        dropIdToUserOrder[_dropId][msg.sender].amountDeposited = uint128(msg.value);
+
+        emit OrderInfo(_dropId, msg.sender, _quantity, uint128(msg.value));
+    }
+
+    function cancelOrder(uint256 _dropId) external {
+        Drop storage drop = drops[_dropId];
+        uint256 amountToReturn = dropIdToUserOrder[_dropId][msg.sender].amountDeposited;
+        require(!isLocked(drop), 'orders are locked for this drop');
+        delete dropIdToUserOrder[_dropId][msg.sender];
+
+        emit OrderInfo(_dropId, msg.sender, 0, 0);
+
+        sendValue(payable(msg.sender), amountToReturn);
+    }
+
+    // HELPERS
+    function isBeforeLockStartTime(Drop storage _drop) private view returns (bool) {
+        return block.timestamp < _drop.lockStartTime;
+    }
+
+    function isLocked(Drop storage _drop) private view returns (bool) {
+        return (block.timestamp > _drop.lockStartTime && block.timestamp < _drop.lockStartTime + _drop.lockLength) ? true : false;
```

Comment:
why not just return `block.timestamp > _drop.lockStartTime && block.timestamp < _drop.lockStartTime + _drop.lockLength` ?